src/Entity/Embargo.php

```diff
@@ -397,29 +397,33 @@ public function getCacheMaxAge() {
    * {@inheritdoc}
    */
   public function getCacheTags() {
-    $tags = Cache::mergeTags(parent::getCacheTags(), $this->getEmbargoedNode()->getCacheTags());
+    if ($this->getEmbargoedNode() !== NULL) {
+      $tags = Cache::mergeTags(parent::getCacheTags(), $this->getEmbargoedNode()->getCacheTags());
 
-    if ($this->getExemptIps()) {
-      $tags = Cache::mergeTags($tags, $this->getExemptIps()->getCacheTags());
+      if ($this->getExemptIps()) {
+        $tags = Cache::mergeTags($tags, $this->getExemptIps()->getCacheTags());
+      }
+      return $tags;
     }
-    return $tags;
   }
 
   /**
    * {@inheritDoc}
    */
   public function getCacheContexts() {
-    $contexts = Cache::mergeContexts(
-      parent::getCacheContexts(),
-      $this->getEmbargoedNode()->getCacheContexts(),
-      ['user.embargo__has_exemption'],
-    );
-
-    if ($this->getExemptIps()) {
-      $contexts = Cache::mergeContexts($contexts, $this->getExemptIps()->getCacheContexts());
+    if ($this->getEmbargoedNode() !== NULL) {
+      $contexts = Cache::mergeContexts(
+        parent::getCacheContexts(),
+        $this->getEmbargoedNode()->getCacheContexts(),
+        ['user.embargo__has_exemption'],
+      );
+
+      if ($this->getExemptIps()) {
+        $contexts = Cache::mergeContexts($contexts, $this->getExemptIps()->getCacheContexts());
+      }
+
+      return $contexts;
     }
-
-    return $contexts;
   }
```

Comment:
Similarly, do rewrite the whole thing to take out one thing.

```php
    $contexts = Cache::mergeContexts(
      parent::getCacheContexts(),
      $this->getEmbargoedNode()->getCacheContexts(),
      ['user.embargo__has_exemption'],
    );
```

can just go to

```php
    $contexts = Cache::mergeContexts(
      parent::getCacheContexts(),
      ['user.embargo__has_exemption'],
    );
    if ($node = $this->getEmbargoedNode()) {
      $context = Cache::mergeContexts($contexts, $node->getCacheContexts());
    }
```

Reply:
Changed.

Reply:
Derp, just realized I typo'd on the assignment of the merged contexts... `s/$context/$contexts/`